nameko/messaging.py

```diff
@@ -157,12 +158,15 @@ def serializer(self):
     def setup(self):
 
         ssl = self.container.config.get(AMQP_SSL_CONFIG_KEY)
+        transport_options = self.container.config.get(
+            TRANSPORT_OPTIONS_CONFIG_KEY, DEFAULT_TRANSPORT_OPTIONS
+        )
```

Comment:
This makes `transport_options` configurable at the service level, rather than the extension level. That might be what we want, but some transport options (e.g. confirms) may want to be different across extensions in the same service.

For the DependencyProviders (`Publisher`, `EventDispatcher` and `RpcProxy`) you could allow `transport_options` to be passed to the constructor (in the same way that `use_confirms` can be) so you could override the default for that particular extension. The corresponding entrypoints don't have the same mechanism for accepting parameters in their constructors, so we'll have to use the service-level options there.

Reply:
Hmm, I can't see it any references to `use_confirms ` in https://github.com/nameko/nameko/blob/master/nameko/messaging.py#L73 except https://github.com/nameko/nameko/blob/master/nameko/messaging.py#L132. 

All `use_confirms` usage seems to be in https://github.com/nameko/nameko/blob/master/nameko/amqp/publish.py#L35 and  I already added transport options https://github.com/nameko/nameko/pull/564/commits/413006b183fdef0b0098771ee5a68428ee2285a6#diff-da23ae33a4d5c536cd0bfb9ca895717aR120.

Reply:
`use_confirms` can be passed into https://github.com/nameko/nameko/blob/master/nameko/messaging.py#L73  as part of ``**options``, and then extracted out of them when that dict is passed to the constructor of https://github.com/nameko/nameko/blob/master/nameko/amqp/publish.py#L35. Actually `transport_options` could already be provided to the DependencyProvider constructor and would also end up in `**options`.

The test at https://github.com/nameko/nameko/blob/0bffbf7b3c2a26ab664075a48d5fc3e136205413/test/test_messaging.py#L1189-L1213 demonstrates this for `use_confirms`. It'd be good to have an equivalent for `transport_options`.

At the moment, the line https://github.com/nameko/nameko/pull/564/files#diff-309aaafd3b61875d021b62c5b3926f91R178 will cause a duplicate keyword argument exception if you did pass `transport_options` to the `Publisher` DependencyProvider constructor. I think if you just removed that line we'd get the desired behaviour.

Does that make sense?